diff --git a/openmmapi/include/openmm/internal/hardware.h b/openmmapi/include/openmm/internal/hardware.h
--- a/openmmapi/include/openmm/internal/hardware.h
+++ b/openmmapi/include/openmm/internal/hardware.h
@@ -94,7 +94,8 @@ static int getNumProcessors() {
 #define cpuid __cpuid
 #else
 #if !defined(__ANDROID__) && !defined(__PNACL__) && !defined(__PPC__) \
-    && !defined(__ARM__) && !defined(__ARM64__) && !defined(__LOONGARCH64__)
+    && !defined(__ARM__) && !defined(__ARM64__) && !defined(__LOONGARCH64__) \
+    && !defined(__EMSCRIPTEN__)
     static void cpuid(int cpuInfo[4], int infoType) {
     #ifdef __LP64__
         __asm__ __volatile__ (
